panic: Failed to create sublogger (smtp): json: cannot unmarshal string into Go struct field SMTPLogger.sendTos of type []string

- Gitea version (or commit ref): 1.10.0
- Operating system: ArchLinux
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

When setting an smtp logger, I get this:

```
panic: Failed to create sublogger (smtp): json: cannot unmarshal string into Go struct field SMTPLogger.sendTos of type []string 
goroutine 1 [running]:
code.gitea.io/gitea/modules/log.NewLogger(0x2710, 0xc0007c13d0, 0x4, 0xc0007c13d0, 0x4, 0xc0004be600, 0xff, 0xc0007abf00)
        code.gitea.io/gitea@/modules/log/log.go:48 +0x1b3
code.gitea.io/gitea/modules/setting.newLogService()
        code.gitea.io/gitea@/modules/setting/log.go:263 +0x7a0
code.gitea.io/gitea/modules/setting.NewLogServices(0xc00182e200)
        code.gitea.io/gitea@/modules/setting/log.go:282 +0x24
code.gitea.io/gitea/modules/setting.NewServices()
        code.gitea.io/gitea@/modules/setting/setting.go:1047 +0x32
code.gitea.io/gitea/routers.NewServices()
        code.gitea.io/gitea@/routers/init.go:44 +0x24
code.gitea.io/gitea/routers.GlobalInit()
        code.gitea.io/gitea@/routers/init.go:79 +0x358
code.gitea.io/gitea/cmd.runWeb(0xc0000f0dc0, 0x0, 0x0)
        code.gitea.io/gitea@/cmd/web.go:108 +0x75
github.com/urfave/cli.HandleAction(0xdf2e200c080, 0xdf2e22a1010, 0xc0000f0dc0, 0xc0000cf300, 0x0)
        github.com/urfave/cli@v1.20.0/app.go:490 +0xca
github.com/urfave/cli.Command.Run(0xdf2df180dcf, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0xdf2df1afbf8, 0x16, 0x0, ...)
        github.com/urfave/cli@v1.20.0/command.go:210 +0x993
github.com/urfave/cli.(*App).Run(0xc0000b9040, 0xc0000d0000, 0x4, 0x4, 0x0, 0x0)
        github.com/urfave/cli@v1.20.0/app.go:255 +0x6ad
main.main()
        code.gitea.io/gitea@/main.go:109 +0x818
```

Under `[log.smtp]`, I’m only doing this:
```
-HOST =
+HOST = 127.0.0.1:25
-RECEIVERS =
+RECEIVERS = someuser@somedomain.com
```

--- a/modules/setting/log.go
+++ b/modules/setting/log.go
@@ -128,7 +128,11 @@ func generateLogConfig(sec *ini.Section, name string, defaults defaultLogOptions
 		logConfig["username"] = sec.Key("USER").MustString("example@example.com")
 		logConfig["password"] = sec.Key("PASSWD").MustString("******")
 		logConfig["host"] = sec.Key("HOST").MustString("127.0.0.1:25")
-		logConfig["sendTos"] = sec.Key("RECEIVERS").MustString("[]")
+		sendTos := strings.Split(sec.Key("RECEIVERS").MustString(""), ",")
+		for i, address := range sendTos {
+			sendTos[i] = strings.TrimSpace(address)
+		}
+		logConfig["sendTos"] = sendTos
 		logConfig["subject"] = sec.Key("SUBJECT").MustString("Diagnostic message from Gitea")
 	}
 

